server/src/main/java/server/formatting/FormattingProvider.java

```diff
@@ -0,0 +1,118 @@
+package server.formatting;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.lsp4j.DocumentFormattingParams;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import server.buildifier.Buildifier;
+import server.buildifier.BuildifierException;
+import server.buildifier.BuildifierFileType;
+import server.buildifier.FormatInput;
+import server.buildifier.FormatOutput;
+import server.utils.DocumentTracker;
+
+public class FormattingProvider {
+
+    private static final Logger logger = LogManager.getLogger(FormattingProvider.class);
+    private DocumentTracker documentTracker;
+    private Buildifier buildifier;
+
+    public FormattingProvider(DocumentTracker documentTracker, Buildifier buildifier) {
+        this.documentTracker = documentTracker;
+        this.buildifier = buildifier;
+    }
+
+
+    public CompletableFuture<List<? extends TextEdit>> getDocumentFormatting(DocumentFormattingParams params) {
+        logger.info("Invoked FormattingProvider");
+
+        FormatInput formatInput = new FormatInput();
+        String stringUri = params.getTextDocument().getUri();
+        logger.info("Formatting file from: " + stringUri);
+        File file = getFileFromUriString(stringUri);
+
+        if (file == null) {
+            // TODO: handle this problem
+            logger.info("Could not find file");
+            return CompletableFuture.completedFuture(new ArrayList<TextEdit>());
+        }
+
+        String name = file.getName();
+        logger.info("Formatting file: " + name);
+
+        if (name.equals("BUILD")) {
+            formatInput.setType(BuildifierFileType.BUILD);
+        } else if (name.equals("WORKSPACE")) {
+            formatInput.setType(BuildifierFileType.WORKSAPCE);
```

Comment:
This supposed to be workspace?

Reply:
This is my B. Yeah we should fix this lol